packages/http-client-csharp/eng/pipeline/publish.yml

```diff
@@ -17,7 +17,7 @@ extends:
         parameters:
           BuildPrereleaseVersion: true
           UseTypeSpecNext: false
-          Publish: "public"
+          Publish: ${{replace(replace('True',ne(variables['Build.Reason'], 'Manual'), 'public'),'True','internal')}}
```

Comment:
I think it is better to change the condition to: if build branch is not main, then set to internal.

Sometimes when publish ci fails, we might want to trigger the ci again manually. In this case, we still want to a public release.

Reply:
you are right! updated